src/blob_gc_job.cc

```diff
@@ -581,5 +581,73 @@ bool BlobGCJob::IsShutingDown() {
   return (shuting_down_ && shuting_down_->load(std::memory_order_acquire));
 }
 
+Status BlobGCJob::DigHole() {
+  Status s;
+  const auto& inputs = blob_gc_->sampled_inputs();
+  assert(!inputs.empty());
+  for (std::size_t i = 0; i < inputs.size(); ++i) {
+    std::unique_ptr<PosixRandomRWFile> file;
+    // TODO(@DorianZheng) set read ahead size
+    s = OpenBlobFile(inputs[i]->file_number(), 0, db_options_, env_options_,
+                     env_, &file);
+    if (!s.ok()) {
+      break;
+    }
+
+    auto gc_iter = new BlobFileIterator(
```

Comment:
gc_iter in previous titan is a merge iter with many files, we should use a better name here.

Reply:
Right. I have renamed to record_iter.